extensions/devservices/keycloak/src/main/java/io/quarkus/devservices/keycloak/KeycloakDevServicesConfig.java

```diff
@@ -188,104 +178,36 @@ public class DevServicesConfig {
      * `user` role.
      * This map is used for role creation when no realm file is found at the `realm-path`.
      */
-    @ConfigItem
     @ConfigDocMapKey("role-name")
-    public Map<String, List<String>> roles;
-
-    /**
-     * Specifies the grant type.
-     *
-     * @deprecated This field is deprecated. Use {@link DevUiConfig#grant} instead.
-     */
-    @Deprecated
-    public Grant grant = new Grant();
-
-    @ConfigGroup
-    public static class Grant {
-        public static enum Type {
-            /**
-             * `client_credentials` grant
-             */
-            CLIENT("client_credentials"),
-            /**
-             * `password` grant
-             */
-            PASSWORD("password"),
-
-            /**
-             * `authorization_code` grant
-             */
-            CODE("code"),
-
-            /**
-             * `implicit` grant
-             */
-            IMPLICIT("implicit");
-
-            private String grantType;
-
-            private Type(String grantType) {
-                this.grantType = grantType;
-            }
-
-            public String getGrantType() {
-                return grantType;
-            }
-        }
-
-        /**
-         * Defines the grant type for aquiring tokens for testing OIDC `service` applications.
-         */
-        @ConfigItem(defaultValue = "code")
-        public Type type = Type.CODE;
-    }
+    Map<String, List<String>> roles();
 
     /**
      * The specific port for the dev service to listen on.
      * <p>
      * If not specified, a random port is selected.
      */
-    @ConfigItem
-    public OptionalInt port;
+    OptionalInt port();
 
     /**
      * Environment variables to be passed to the container.
      */
-    @ConfigItem
     @ConfigDocMapKey("environment-variable-name")
-    public Map<String, String> containerEnv;
+    Map<String, String> containerEnv();
 
     /**
      * Memory limit for Keycloak container
      * </p>
      * If not specified, 750MiB is the default memory limit.
      */
-    @ConfigItem(defaultValue = "750M")
-    public MemorySize containerMemoryLimit;
+    @WithDefault("750M")
+    MemorySize containerMemoryLimit();
 
-    @Override
```

Comment:
Just commenting that that this was done to check if Keycloak devservice may have to be reloaded, I'm yet to check the moved KC devservice code, so not sure yet these checks have moved there

Reply:
Sure thing, look here https://smallrye.io/smallrye-config/Main/config/mappings/#tostring-equals-hashcode.

Reply:
Just for context - I simply removed it as SR Config generates it for us.

Reply:
@michalvavrik Cool